### PR TITLE
docs: update solution overview

### PR DESCRIPTION
**Pull request by swiatekm-sumo**
Update the solution overview. Remove the old diagram, remove history from the supported software version table. Add new diagrams.

We're missing a diagram for traces, but I'd like to add that separately.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -92,41 +77,18 @@ The following table displays the tested Kubernetes and Helm versions.
...
-| 0.12.0                | 8.2.0                                     | 1.6.3   | 2.8.1                               | 1.0.9  | -              | -                 |                          |                        |
-| 0.9.0 - 0.11.0        | 6.2.1                                     | 1.6.3   | 2.4.4                               | 1.0.8  | -              | -                 |                          |                        |
-| 0.6.0 - 0.8.0         | 6.2.1                                     | 1.6.3   | 2.4.4                               | 1.0.5  | -              | -                 |                          |                        |
+The following table displays the currently used software versions for our Helm chart.
+
+| Name                                      | Version |
+|-------------------------------------------|---------|
+| OpenTelemetry Collector                   | 0.66.0  |
+| OpenTelemetry Operator                    | 0.18.3  |
+| kube-prometheus-stack/Prometheus Operator | 42.1.0  |
```
Comment (kkujawa-sumo) [inline docs/README.md:86]: Please change this to currently used version of kube-prometheus-stack https://github.com/SumoLogic/sumologic-kubernetes-collection/blob/7d91421f420c4a06045f9f6ff9efd7972c2331a1/deploy/helm/sumologic/Chart.yaml#L20

Comment (swiatekm-sumo) [reply docs/README.md]: Done.

File: docs/README.md [hunk 2]
```diff
@@ -92,41 +77,18 @@ The following table displays the tested Kubernetes and Helm versions.
...
+
+| Name                                      | Version |
+|-------------------------------------------|---------|
+| OpenTelemetry Collector                   | 0.66.0  |
+| OpenTelemetry Operator                    | 0.18.3  |
+| kube-prometheus-stack/Prometheus Operator | 40.5.0  |
+| Falco                                     | 2.4.2   |
+| Telegraf Operator                         | 1.3.10  |
+| Tailing Sidecar Operator                  | 0.5.5   |
+| FluentD                                   | 1.14.6  |
```
Comment (astencel-sumo) [inline docs/README.md:90]:
```suggestion
| Fluentd                                   | 1.14.6  |
```

File: docs/README.md [hunk 3]
```diff
@@ -92,41 +77,18 @@ The following table displays the tested Kubernetes and Helm versions.
...
+| Fluent Bit                                | 1.6.10  |
```
Comment (sumo-drosiek) [inline docs/README.md:91]: Perfect case for automation/ci check :star_struck:

Comment (swiatekm-sumo) [reply docs/README.md:91]: It is indeed.

File: docs/README.md (hunk 2 above)
Comment (swiatekm-sumo) [reply docs/README.md]: 1.15.3 now as well, fixed it.

File: docs/README.md [hunk 4]
```diff
@@ -54,28 +54,13 @@ Documentation links:
...
-- **kube-state-metrics.** Listens to the Kubernetes API server; generates metrics
-  about the state of the deployments, nodes, and pods in the cluster; and exports
-  the metrics as plaintext on an HTTP endpoint listen port.
-- **Prometheus deployment.** Scrapes the metrics exposed by the `node-exporter`
-  add-on for Kubernetes and the `kube-state-metrics` component; writes metrics
-  to a port on the Fluentd deployment.
-- **Fluentd deployment.** Forwards logs and metrics to HTTP sources on a hosted collector.
-  Includes multiple Fluentd plugins that parse and format the metrics and enrich them with metadata.
-- **Events Fluentd deployment.** Forwards events to an HTTP source on a hosted collector.
+:construction: TODO: Add a new diagram here
```
Comment (sumo-drosiek) [inline docs/README.md:57]: Please remove old image if we don't use it

**Comment by swiatekm-sumo** [comment]
> @swiatekm-sumo Why do you remove [docs/additional-prometheus-configuration.md](https://github.com/SumoLogic/sumologic-kubernetes-collection/pull/2718/files#diff-8b368fc211d0922215037ff63ef04b5b605025c5c2a9b274a0e3b3a16ae81cba) in this PR?

My bad, that was just a messy rebase.

**Review by sumo-drosiek — COMMENTED** [review]
`pull metrics` is too low under the arrow in comparison to other arrows

File: docs/README.md [hunk 5]
```diff
@@ -52,30 +52,21 @@ Documentation links:
...
-  add-on for Kubernetes and the `kube-state-metrics` component; writes metrics
-  to a port on the Fluentd deployment.
-- **Fluentd deployment.** Forwards logs and metrics to HTTP sources on a hosted collector.
-  Includes multiple Fluentd plugins that parse and format the metrics and enrich them with metadata.
-- **Events Fluentd deployment.** Forwards events to an HTTP source on a hosted collector.
+The diagrams below illustrate the components of the Kubernetes collection solution.
+
+### Log Collection
+
+![logs](/images/logs.png)
```
Comment (astencel-sumo) [inline docs/README.md:59]: The Kubernetes node's icon and the Kubernetes API server's icon should have a title, like "k8s node" or "node" and "api-server" or similar.

Comment (astencel-sumo) [inline docs/README.md:59]: I would rather have the "disk" icon at the bottom of the k8s node, and pods and OTC above it. :zany_face:

**Comment by astencel-sumo** [comment]
Is it possible to have a "source code" for these images available? How can we make changes to these images if needed in the future?

**Comment by astencel-sumo** [comment]
If I understand correctly, this PR adds diagrams for logs and metrics, but not for traces and not for events. Is this intended?

**Comment by swiatekm-sumo** [comment]
> Is it possible to have a "source code" for these images available? How can we make changes to these images if needed in the future?



> If I understand correctly, this PR adds diagrams for logs and metrics, but not for traces and not for events. Is this intended?

Good point about events, I'll add that. Traces weren't present originally, and I'd like to add them separately.

File: docs/README.md (hunk 5 above)
Comment (swiatekm-sumo) [reply docs/README.md:59]: I tried that, and it didn't really look better imo. Is this what you have in mind?
Screenshot_20230104_122213

Comment (swiatekm-sumo) [reply docs/README.md:59]: Are you sure it's necessary for the api server? The icon even includes the word "api".

**Comment by swiatekm-sumo** [comment]
Added the events diagram and the draw.io sources for the diagrams. This is now ready for final review.

**Review by sumo-drosiek — COMMENTED** [review]
I don't like the one side arrows, I would make it dotted line for pull data/api request and regular line for data flow. Except that LGTM

**Comment by swiatekm-sumo** [comment]
> I don't like the one side arrows, I would make it dotted line for pull data/api request and regular line for data flow. Except that LGTM

I think it's fine with arrow labels clarifying what they mean. Adding more arrows just makes the diagrams more difficult to read. To me, part of the point of these is to illustrate the interaction model (push vs pull), and doing what you suggest will be confusing once I also add a traces diagram with arrows pointing the other way.

**Comment by sumo-drosiek** [comment]
> > I don't like the one side arrows, I would make it dotted line for pull data/api request and regular line for data flow. Except that LGTM
> 
> I think it's fine with arrow labels clarifying what they mean. Adding more arrows just makes the diagrams more difficult to read. To me, part of the point of these is to illustrate the interaction model (push vs pull), and doing what you suggest will be confusing once I also add a traces diagram with arrows pointing the other way.

I don't know. It says pull metadata, but the arrow suggest that the data flow is from opentelemetry to API :upside_down_face:

**Comment by swiatekm-sumo** [comment]
> > > I don't like the one side arrows, I would make it dotted line for pull data/api request and regular line for data flow. Except that LGTM
> > 
> > 
> > I think it's fine with arrow labels clarifying what they mean. Adding more arrows just makes the diagrams more difficult to read. To me, part of the point of these is to illustrate the interaction model (push vs pull), and doing what you suggest will be confusing once I also add a traces diagram with arrows pointing the other way.
> 
> I don't know. It says pull metadata, but the arrow suggest that the data flow is from opentelemetry to API upside_down_face

It's not intended to be data flow, it's intended to be an illustration of interactions between components. Otel pulls data from the API Server, therefore the arrow goes that way.

**Pull request merged**